File: src/main/java/com/key/oa/service/impl/EmployeeServiceImpl.java
package com.key.oa.service.impl;

import com.key.oa.entity.Employee;
import com.key.oa.repository.EmployeeRepository;
import com.key.oa.service.EmployeeService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * @author 孙强
 * 员工类的Service实现类
 */
@Service
public class EmployeeServiceImpl implements EmployeeService {
    private final EmployeeRepository repository;

    @Autowired
    public EmployeeServiceImpl(EmployeeRepository employeeRepository) {
        this.repository = employeeRepository;
    }

    @Override
    public Page<Employee> findAll(Pageable pageable) {
        return repository.findAll(pageable);
    }

    @Override
    public Employee findByJobNumber(String jobNumber) {
        return repository.findByJobNumber(jobNumber);
    }

    @Override
    public Employee save(Employee employee) {
        // 对密码进行加密，初始密码是1234
        employee.setPassword(new BCryptPasswordEncoder().encode("1234"));
        // 新建员工要设置成“未激活”状态
        employee.setVerified(false);
        return repository.save(employee);
    }

    @Override
    public Employee update(Employee employee) {
        return repository.save(employee);
    }

    @Override
    public void deleteById(Long id) {
        repository.deleteById(id);
    }

    @Override
    public long count() {
        return repository.count();
    }

    @Override
    @Transactional(rollbackFor = Exception.class)
    public void deleteBatch(List<String> jobNumberList) {
        List<Employee> employeeList = repository.findAllByJobNumberIn(jobNumberList);
        if (employeeList != null) {
            repository.deleteAllInBatch(employeeList);
        }
    }
}

File: src/main/java/com/key/oa/service/impl/EmployeeServiceImpl.kt
package com.key.oa.service.impl

import com.key.oa.entity.Employee
import com.key.oa.repository.EmployeeRepository
import com.key.oa.service.EmployeeService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class EmployeeServiceImpl @Autowired constructor(private val repository: EmployeeRepository) : EmployeeService {
    override fun findAll(pageable: Pageable): Page<Employee> = repository.findAll(pageable)

    override fun findByJobNumber(jobNumber: String): Employee? = repository.findByJobNumber(jobNumber)

    override fun save(employee: Employee): Employee {
        // 设置初始密码为'1234'，状态为'未激活'
        employee.password = "1234"
        employee.verified = false

        return repository.save(employee)
    }

    override fun update(employee: Employee): Employee = repository.save(employee)

    override fun deleteById(id: Long) = repository.deleteById(id)

    override fun count(): Long = repository.count()

    @Transactional(rollbackFor = [Exception::class])
    override fun deleteBatch(jobNumberList: List<String>) {
        val employeeList = repository.findAllByJobNumberIn(jobNumberList)
        if (employeeList != null) {
            repository.deleteAllInBatch(employeeList)
        }
    }
}